libcore/ConnectServerDialog.vala

```diff
@@ -130,8 +127,9 @@ public class PF.ConnectServerDialog : Granite.Dialog {
 
         var server_header_label = new Granite.HeaderLabel (_("Server Details"));
 
-        server_entry = new DetailEntry (_("Server name or IP address")) {
-            hexpand = true
+        server_entry = new Granite.ValidatedEntry () {
+            hexpand = true,
+            placeholder_text = _("Server name or IP address")
         };
 
```

Comment:
You can ensure the placeholder is fully visible by adding 
```
server_entry.width_chars = server_entry.placeholder_text.length;
```

or to ensure long enough to show a full IP address (at least IPv4 - I can't see anyone entering an IPv6 address manually)
```
server_entry.width_chars = int.max (server_entry.placeholder_text.length, 15);
```

Reply:
I'm not sure if it's a good idea to use the placeholder width chars here since that might end up being a very long string in some more verbose locales and users can now resize the dialog if they need to do that. Setting a min width for an expended IP Address length could work though. I think what might end up being best in the long run is to change this placeholder text to something like: `example.com or 104.21.3.130`